content/blog/2025-10-20-zenoh-imoogi.md

```diff
@@ -0,0 +1,457 @@
+# HELP tx_n_msgs Counter of sent network messages.
+# TYPE tx_n_msgs counter
+tx_n_msgs 65
+tx_n_msgs{media="net"} 20
+tx_n_msgs{media="shm"} 45
+```
+
+Note: to enable statistics, Zenoh has to be built with the `stats` feature.
+
+## Configuration changes
+
+In version 1.6.1, we introduced minor modifications to the Zenoh configuration to improve its overall consistency:
+
+* The unstable `congestion_control` value "blockfirst" was renamed to "block_first".
+* The "downsampling/messages" value "push" has been deprecated and may no longer be supported in future versions. It has been replaced with "put" and "delete" values (similar to other interceptors), which allow separate control of the publishing frequency for PUT and DELETE messages.
+
+## Scalability improvements
+
+In version 1.6.1, we fixed a bug that was causing an infinite loop of messages across multiple processes in a peer-to-peer topology [(PR)](https://github.com/eclipse-zenoh/zenoh/pull/214). We also applied various optimizations that reduce CPU consumption for discovery message processing [(PR)](https://github.com/eclipse-zenoh/zenoh/pull/2174).
+
+Overall, these changes drastically reduced CPU consumption and significantly improved Zenoh scalability, especially in peer-to-peer scenarios and the rmw_zenoh use case.
+
+## Rust 1.75 compatibility improvement
+
+The **zenoh** crate is declared to be compatible with Rust 1.75. Unfortunately, the Rust version compatibility checker ignores the fact that dependent crates evolve, and a crate that worked with Rust 1.75 at one point may later fail to compile because some of its dependencies have bumped their minimum supported versions.
+
+One possible solution would be to use pinned dependencies (e.g., `"=0.1.2"`) in zenoh itself, but this approach can cause compatibility issues.
+
+The solution is to include an additional crate, [zenoh-pinned-deps-1-75](https://crates.io/crates/zenoh-pinned-deps-1-75), which pins all failing dependencies (as of the time of release 😭) to versions compatible with Rust 1.75.
+
+## Documentation improvements
+
+The [Rust documentation](https://docs.rs/zenoh/latest/zenoh/) has been significantly extended. Introductory paragraphs with usage examples were added to the main page and each module. Features were documented, and functions and types were cross-linked to ensure that users would not get lost when encountering a type without knowing its purpose or how to use it.
+
+The project's [README](https://github.com/eclipse-zenoh/zenoh/blob/main/README.md) was also improved and systematized. Each component now has its own README, with the root README linking to all of them.
+
+# Zenoh-Pico
+
+## Advanced Pub/Sub
+
+Zenoh-Pico now includes support for the Advanced Publisher and Subscriber, bringing it in line with core Zenoh, where this functionality was first introduced in version [1.1.0](https://zenoh.io/blog/2024-12-12-zenoh-firesong-1.1.0/).
+
+This version introduces:
+
+* An Advanced Publisher that can:
```

Comment:
Worth mentioning that this can be used to provide reliable communication when using best effort links such as UDP/IP